doc/api/buffer.md

```diff
@@ -279,6 +279,101 @@ for (const b of buf) {
 Additionally, the [`buf.values()`][], [`buf.keys()`][], and
 [`buf.entries()`][] methods can be used to create iterators.
 
+## Class: `Blob`
+<!-- YAML
+added: REPLACEME
+-->
+
+A `Blob` encapsulates immutable, raw data.
+
+### `new buffer.Blob(sources[, options])`
+<!-- YAML
+added: REPLACEME
+-->
+
+* `sources` {string[]|ArrayBuffer[]|TypedArray[]|DataView[]|Blob[]} An array
+  of string, {ArrayBuffer}, {TypedArray}, {DataView}, or {Blob} objects, or
+  any mix of such objects, that will be stored within the `Blob`.
+* `options` {Object}
+  * `encoding` {string} The character encoding to use for string sources.
+    **Default**: `'utf8'`.
+  * `type` {string} The Blob content-type.
+
+Creates a new `Blob` object containing a concatenation of the given sources.
+The sources are copied into the 'Blob' and can therefore be safely modified
+after the 'Blob' is created.
+### `blob.arrayBuffer()`
+<!-- YAML
+added: REPLACEME
+-->
+
+* Returns: {Promise}
+
+Returns a promise that resolves with an {ArrayBuffer} containing a copy of
+the `Blob` data.
+
+### `blob.size`
+<!-- YAML
+added: REPLACEME
+-->
+
+The total size of the `Blob` in bytes.
+
+### `blob.slice([start, [end, [type]]])`
+<!-- YAML
+added: REPLACEME
+-->
+
+* `start` {number} The starting index.
+* `end` {number} The ending index.
+* `type` {string} The content-type for the new `Blob`
+
+Creates and returns a new `Blob` containing a subset of this `Blob' objects
+data. The original `Blob` is not alterered.
+
+### `blob.text()`
+<!-- YAML
+added: REPLACEME
+-->
+
+* Returns: {Promise}
+
+Returns a promise that resolves the contents of the `Blob` encoded as a UTF-8
+string.
+
+### `blob.type`
+<!-- YAML
+added: REPLACEME
+-->
+
+* Type: {string}
+
+The content-type of the `Blob`.
+
+### `Blob` objects and `MessageChannel`
+
+{Blob} objects can be sent using a `MessagePort`. The advantage of sending a
```

Comment:
I don't understand why that's an advantage? I'd understand CoW or something but why would you send the blob over a message port if you don't intend to read it?

Reply:
I’m also a bit doubtful that we can actually achieve this, tbh. If one wants zero-copy behavior, I think the only realistic & spec-compliant way to get that is to just use `SharedArrayBuffer`.

Reply:
The semantics here are Deferred-copy, not Zero-copy. When a `Blob` is created there's an initial copy into the immutable store. References to that can then be copied around to any thread without incurring the cost of an additional copy immediately on send. The advantage here is not that I don't intend to read it, it's that I want to defer the copy until when I want to read it.

* `SharedArrayBuffer` == Zero-Copy, Mutable Shared Memory
* `ArrayBuffer` == Immediate Copy or Transferred, Mutable Exclusive Memory
* `Blob` == Deferred-Copy, Immutable Shared Memory

One could implement an immutable wrapper around `SharedArrayBuffer` to accomplish the same deferred-copy semantics but that wasn't the only goal here. An implementation of the web platform Blob API and eventually File/FileReader and fetch are eventual goals.

Reply:
I would reword the docs then, as it’s somewhat misleading right now – there is no actual performance advantage here if we have to do a copy operation at creation time, there’s only the advantage of being able to copy it multiple times to different destinations.

> One could implement an immutable wrapper around `SharedArrayBuffer` to accomplish the same deferred-copy semantics but that wasn't the only goal here.

Well … it would enable us to implement this in pure JS, which seems like it would be a bit nicer?

Reply:
Updated the docs. Hopefully it's clearer